packages/contracts/test/test-utils/interfaces.ts

```diff
@@ -0,0 +1,11 @@
+import { ethers as ethersDirect } from 'ethers';
+import { ethers } from 'hardhat';
+
+export function getInterfaceID(contractInterface: ethersDirect.utils.Interface) {
+    let interfaceID = ethers.constants.Zero;
+    const functions: string[] = Object.keys(contractInterface.functions);
+    for (let i = 0; i < functions.length; i++) {
+        interfaceID = interfaceID.xor(contractInterface.getSighash(functions[i]));
+    }
+    return interfaceID.toHexString();
+}    
```

Comment:
new line at the end of file.

Reply:
Please use prettier as the default formatter.
In visual studio code, press `cmd` + `shift` + `p`, then type `Format Document With ...`, then chose `Configure Default Formatter`, and then `Prettier - Code Formatter`.
This requires the [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extension